stan/math/rev/fun/elt_multiply.hpp

```diff
@@ -34,10 +34,15 @@ auto elt_multiply(const Mat1& m1, const Mat2& m2) {
     arena_t<promote_scalar_t<var, Mat2>> arena_m2 = m2;
     arena_t<ret_type> ret(arena_m1.val().cwiseProduct(arena_m2.val()));
     reverse_pass_callback([ret, arena_m1, arena_m2]() mutable {
-      auto ret_adj = ret.adj().array();
-      eigen_results(arena_m1.adj().array(), arena_m2.adj().array())
-          += eigen_expressions(arena_m2.val().array() * ret_adj,
-                               arena_m1.val().array() * ret_adj);
+      if (is_var_matrix<Mat1>::value || is_var_matrix<Mat2>::value) {
+        arena_m1.adj() += arena_m2.val().cwiseProduct(ret.adj());
+        arena_m2.adj() += arena_m1.val().cwiseProduct(ret.adj());
```

Comment:
I doubt this will be any faster, its just more code to do the same thing. I want to see a benchmark showing that this is faster.

Reply:
We've formally entered the Age of Benchmarking. Exciting!

Reply:
You were right this was much worse! Forgot to save the graph but this was way slower